docs/src/handle.rst

```diff
@@ -140,6 +140,8 @@ API
 
     Request handle to be closed. `close_cb` will be called asynchronously after
     this call. This MUST be called on each handle before memory is released.
+    Moreover, the memory can only be released at the end of `close_cb` or after
+    it has returned.
```

Comment:
"at the end" isn't quite correct. It doesn't matter if you do it at the start, end or afterwards, as long as you don't do it _before_ the close callback is called.